File: csharp/src/ev1/ut3_strings/ejercicios/resueltos/E09_InvertirCadena_Resuelto.cs
/*
OBJETIVO: Lee una cadena e imprime la misma cadena invertida con un bucle.
SOLUCION: ver codigo. Se presenta con menu do-while y opcion 0 para salir.

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

string opcion;

do
{
    Console.WriteLine("=== E09 Invertir cadena (resuelto) ===");
    Console.WriteLine("1. Invertir cadena");
    Console.WriteLine("2. Invertir y comparar con original");
    Console.WriteLine("0. Salir");
    Console.Write("Elige una opcion: ");
    opcion = Console.ReadLine() ?? "0";
    Console.WriteLine();

    switch (opcion)
    {
        case "1":
        case "2":
            Console.Write("Escribe una cadena: ");
            string texto = Console.ReadLine() ?? "";
            string invertida = "";

            for (int i = texto.Length - 1; i >= 0; i--)
            {
                invertida += texto[i];
            }

            Console.WriteLine("Invertida: " + invertida);

            if (opcion == "2")
            {
                Console.WriteLine(texto == invertida ? "Coincide con la original." : "No coincide con la original.");
            }
            break;

        case "0":
            Console.WriteLine("Saliendo del ejercicio.");
            break;

        default:
            Console.WriteLine("Opcion no valida.");
            break;
    }

    Console.WriteLine();
} while (opcion != "0");

File: cpp/src/ev1/ut3_strings/ejercicios/resueltos/E09_InvertirCadena_Resuelto.cpp
/*
OBJETIVO: Lee una cadena e imprime la misma cadena invertida con un bucle.
SOLUCION: ver codigo. Se presenta con menu do-while y opcion 0 para salir.

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

#include <iostream>
#include <string>
using namespace std;

int main() {
    string opcion;

    do {
        cout << "=== E09 Invertir cadena (resuelto) ===" << endl;
        cout << "1. Invertir cadena" << endl;
        cout << "2. Invertir y comparar con original" << endl;
        cout << "0. Salir" << endl;
        cout << "Elige una opcion: ";
        getline(cin, opcion);
        cout << endl;

        if (opcion == "1" || opcion == "2") {
            cout << "Escribe una cadena: ";
            string texto;
            getline(cin, texto);
            string invertida = "";

            for (int i = (int)texto.size() - 1; i >= 0; i--) {
                invertida += texto[i];
            }

            cout << "Invertida: " << invertida << endl;

            if (opcion == "2") {
                if (texto == invertida) {
                    cout << "Coincide con la original." << endl;
                } else {
                    cout << "No coincide con la original." << endl;
                }
            }
        } else if (opcion == "0") {
            cout << "Saliendo del ejercicio." << endl;
        } else {
            cout << "Opcion no valida." << endl;
        }

        cout << endl;
    } while (opcion != "0");

    return 0;
}
